gateways/clients/wallet_service_db_client.py

```diff
@@ -58,6 +57,12 @@
 WHERE EXISTS (SELECT 1 FROM address_balance WHERE address = :address and token_id = '00')
 '''
 
+address_tokens_count_query = '''\
+SELECT COUNT(*) as total
+FROM token INNER JOIN address_balance
+ON token.id = address_balance.token_id
+WHERE address_balance.address = :address'''
```

Comment:
I could query only on `address_balance` since we only need to count distincts `token_id` for an address.
I prefer querying with the `inner join` to make sure the response here will always return the same number of tokens that can be found with the `address_tokens_query` (plus HTR).

The address_balance and token tables should never be "out of sync" but I had this issue on my local test (which is not fully synced).